packages/workbox-routing/src/lib/router.js

```diff
@@ -46,113 +51,152 @@ import normalizeHandler from './normalize-handler';
  * });
  *
  * const router = new workbox.routing.Router();
+ * router.addFetchListener();
  * router.registerRoutes({routes: [assetRoute, imageRoute]});
  * router.setDefaultHandler({
  *   handler: new workbox.runtimeCaching.NetworkFirst(),
  * });
  */
 class Router {
   /**
-   * Start with an empty array of routes, and by default, sets up the `fetch`
-   * handler.
-   *
-   * @param {Boolean} [handleFetch=true] Whether or not the Router should
-   * register a `fetch` handler and respond to requests. This can be set to
-   * `false` in a development environment, to prevent the service worker
-   * from responding with cached responses.
+   * Constructs a new instance, without any registered routes.
    */
-  constructor({handleFetch} = {}) {
-    if (typeof handleFetch === 'undefined') {
-      handleFetch = true;
-    }
-
+  constructor() {
     // _routes will contain a mapping of HTTP method name ('GET', etc.) to an
     // array of all the corresponding Route instances that are registered.
     this._routes = new Map();
+  }
+
+  /**
+   * This will register a `fetch` event listener on your behalf which will check
+   * the incoming request to see if there's a matching route, and only respond
+   * if there is a match.
+   *
+   * @example
+   * const imageRoute = new RegExpRoute({
+   *   regExp: /images/,
+   *   handler: new CacheFirst(),
+   * });
+   *
+   * const router = new Router();
+   * router.registerRoute({route: imageRoute});
+   * router.addFetchListener();
+   */
+  addFetchListener() {
+    self.addEventListener('fetch', (event) => {
+      const responsePromise = this.handleRequest({event});
```

Comment:
The latter.

`waitUntil` is only needed if there are multiple asynchronous things that are part of separate promise chains going on inside the event handler. When there's just a single promise chain that resolves with the `Response`, it can be passed directly to `respondWith()`.